### PR TITLE
python3Packages.cxxheaderparser: init at 1.3.3

### PR DESCRIPTION
**Pull request by nim65s**
## Description of changes

Add https://github.com/robotpy/cxxheaderparser, a Modern pure python C++ header parser 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 333499` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cxxheaderparser</li>
    <li>python311Packages.cxxheaderparser.dist</li>
    <li>python312Packages.cxxheaderparser</li>
    <li>python312Packages.cxxheaderparser.dist</li>
  </ul>
</details>

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 333499` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cxxheaderparser</li>
    <li>python311Packages.cxxheaderparser.dist</li>
    <li>python312Packages.cxxheaderparser</li>
    <li>python312Packages.cxxheaderparser.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/cxxheaderparser/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  setuptools,
+  wheel,
```
Comment (dotlambda) [inline pkgs/development/python-modules/cxxheaderparser/default.nix:6]:
```suggestion
```

File: pkgs/development/python-modules/cxxheaderparser/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  postPatch = ''
+    # version.py is generated based on latest git tag
+    echo "__version__ = '${version}'" > cxxheaderparser/version.py
+  '';
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
```
Comment (dotlambda) [inline pkgs/development/python-modules/cxxheaderparser/default.nix:28]:
```suggestion
```

File: pkgs/development/python-modules/cxxheaderparser/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    rev = version;
+    hash = "sha256-dc+MsSJFeRho6DG21QQZSAq4PmRZl7OlWhIQNisicZo=";
+  };
+
+  postPatch = ''
+    # version.py is generated based on latest git tag
+    echo "__version__ = '${version}'" > cxxheaderparser/version.py
+  '';
+
+  nativeBuildInputs = [
```
Comment (dotlambda) [inline pkgs/development/python-modules/cxxheaderparser/default.nix:26]:
```suggestion
  build-system = [
```

**Review by dotlambda — COMMENTED** [review]
Please add `nativeCheckInputs`.

**Comment by nim65s** [comment]
Thanks for the review !

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 333499` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cxxheaderparser</li>
    <li>python311Packages.cxxheaderparser.dist</li>
    <li>python312Packages.cxxheaderparser</li>
    <li>python312Packages.cxxheaderparser.dist</li>
  </ul>
</details>

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 333499` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cxxheaderparser</li>
    <li>python311Packages.cxxheaderparser.dist</li>
    <li>python312Packages.cxxheaderparser</li>
    <li>python312Packages.cxxheaderparser.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
@ofborg build python312Packages.cxxheaderparser

**Review by natsukium — APPROVED** [review]

**Pull request merged**